src/menu/stateful-container.js

```diff
@@ -149,11 +154,70 @@ export default class MenuStatefulContainer extends React.Component<
         }
         this.handleEnterKey(event);
         break;
+      default:
+        if (this.props.typeAhead) this.findItemToFocus(event);
+        break;
     }
   };
+  findItemToFocus = (event: KeyboardEvent) => {
```

Comment:
An idiomatic way to handle the timeout would be the example below

```js
class MyComponent {

  typeaheadCharacters = '';
  typeaheadTimeout = null;

  handleKeyDown = (event) => {
    // ignoring arrow keys
    clearTimeout(this.typeaheadTimeout);
    this.handleAlphaDown(event);
  }

  handleAlphaDown = (event) => {
    this.typeaheadCharacters += event.key;
    this.typeaheadTimeout = setTimeout(() => {
      this.typeaheadCharacters = '';
    }, 500);

    // search for the matching item here
  }
}
```

Reply:
Yeah. Thanks!